OsmAnd/src/net/osmand/plus/mapmarkers/ItineraryDataHelper.java

```diff
@@ -0,0 +1,421 @@
+					ItineraryGroupInfo groupInfo = new ItineraryGroupInfo();
+
+					int tok;
+					while ((tok = parser.next()) != XmlPullParser.END_DOCUMENT) {
+						if (tok == XmlPullParser.START_TAG) {
+							String tagName = parser.getName().toLowerCase();
+							if ("name".equals(tagName)) {
+								groupInfo.name = readText(parser, tagName);
+							} else if ("type".equals(tagName)) {
+								groupInfo.type = readText(parser, tagName);
+							} else if ("path".equals(tagName)) {
+								groupInfo.path = readText(parser, tagName);
+							} else if ("alias".equals(tagName)) {
+								groupInfo.alias = readText(parser, tagName);
+							} else if ("categories".equals(tagName)) {
+								groupInfo.categories = readText(parser, tagName);
+							}
+						} else if (tok == XmlPullParser.END_TAG) {
+							if (ITINERARY_GROUP.equalsIgnoreCase(parser.getName())) {
+								groupInfos.add(groupInfo);
+								return true;
+							}
+						}
+					}
+				}
+				return false;
+			}
+		});
+	}
+
+	public String saveMarkersToFile(String fileName) {
+		GPXFile gpxFile = generateGpx();
+		String dirName = IndexConstants.GPX_INDEX_DIR + IndexConstants.MAP_MARKERS_INDEX_DIR;
+		File dir = app.getAppPath(dirName);
+		if (!dir.exists()) {
+			dir.mkdirs();
+		}
+		String uniqueFileName = FileUtils.createUniqueFileName(app, fileName, dirName, IndexConstants.GPX_FILE_EXT);
+		File fout = new File(dir, uniqueFileName + IndexConstants.GPX_FILE_EXT);
+		GPXUtilities.writeGpxFile(fout, gpxFile);
+
+		return fout.getAbsolutePath();
+	}
+
+	public GPXFile generateGpx() {
+		return generateGpx(mapMarkersHelper.getMapMarkers(), false);
+	}
+
+	public GPXFile generateGpx(List<MapMarker> markers, boolean completeBackup) {
+		GPXFile gpxFile = new GPXFile(Version.getFullVersion(app));
+		for (MapMarker marker : markers) {
+			WptPt wpt = toWpt(marker);
+			wpt.setColor(ContextCompat.getColor(app, MapMarker.getColorId(marker.colorIndex)));
+			if (completeBackup) {
+				if (marker.creationDate != 0) {
+					wpt.getExtensionsToWrite().put(CREATION_DATE, GPX_TIME_FORMAT.format(new Date(marker.creationDate)));
+				}
+				if (marker.visitedDate != 0) {
+					wpt.getExtensionsToWrite().put(VISITED_DATE, GPX_TIME_FORMAT.format(new Date(marker.visitedDate)));
+				}
+			}
+			gpxFile.addPoint(wpt);
+		}
+		return gpxFile;
+	}
+
+	public GPXFile generateGpx(List<MapMarkersGroup> mapMarkersGroups) {
+		GPXFile gpxFile = new GPXFile(Version.getFullVersion(app));
+		List<ItineraryGroupInfo> groups = new ArrayList<>();
+		for (MapMarkersGroup group : mapMarkersGroups) {
+			ItineraryGroupInfo groupInfo = ItineraryGroupInfo.createGroupInfo(app, group);
+
+			for (MapMarker marker : group.getMarkers()) {
+				WptPt wptPt = toWpt(marker);
+
+				Map<String, String> extensions = wptPt.getExtensionsToWrite();
+
+				if (group.getType() == ItineraryType.MARKERS) {
+					extensions.put(ITINERARY_ID, groupInfo.alias + ":" + marker.id);
```

Comment:
check alias don't have ':'
check that id is parsed correctly if name of the point also has ':'